Microsoft.Toolkit.Forms.UI.XamlHost/WindowsXamlHostBase.cs

```diff
@@ -20,38 +18,45 @@ namespace Microsoft.Toolkit.Forms.UI.XamlHost
     [System.ComponentModel.DesignerCategory("code")]
     public abstract partial class WindowsXamlHostBase : ContainerControl
     {
-#pragma warning disable CA1051 // Do not declare visible instance fields
-#pragma warning disable SA1401 // Fields must be private
         /// <summary>
-        /// DesktopWindowXamlSource instance
+        /// UWP XAML Application instance and root UWP XamlMetadataProvider.  Custom implementation required to
+        /// probe at runtime for custom UWP XAML type information.  This must be created before
+        /// creating any DesktopWindowXamlSource instances if custom UWP XAML types are required.
         /// </summary>
-        protected internal readonly windows.UI.Xaml.Hosting.DesktopWindowXamlSource _xamlSource;
+        /// <remarks>
+        /// <seealso cref="WUX.Application"/> object is required for loading custom control metadata.  If a custom
+        /// Application object is not provided by the application, the host control will create one (XamlApplication).
```

Comment:
nit: Consider updating this comment now that custom metadata providers can be added after Xaml.Application creation, if the WindowsXamlHost Application object is used.  Should users ever care about their own custom Xaml.Application object now that they have this flexibity?

Reply:
We put a sample code in the WPF App in the declaration of MyControl

---
In reply to: [261451447](https://github.com/windows-toolkit/Microsoft.Toolkit.Win32/pull/85#discussion_r261451447) [](ancestors = 261451447)